### PR TITLE
Add functions to allow storage of bitarrays

### PR DESCRIPTION
**Pull request by evanh**
Add functions that allow bitarrays to be stored in offline files
or databases.

### CODE REVIEW

File: bitarray/sparse_bitarray.go [hunk 1]
```diff
@@ -296,6 +300,67 @@ func (sba *sparseBitArray) IntersectsBetween(other BitArray, start, stop uint64)
 	return true
 }
 
+// WriteSparse serializes the sparseBitArray and passes it to the writer.
+func WriteSparse(w io.Writer, ba *sparseBitArray) error {
```
Comment (alexandercampbell-wf) [inline bitarray/sparse_bitarray.go]: This exported function has an unexported type in the argument list. This unusable from an external library. I'm unsure of how to solve this. I don't think it's ideal to export `sparseBitArray`.

@dustinhiatt-wf @tylertreat-wf thoughts?

Comment (tylertreat-wf) [inline bitarray/sparse_bitarray.go]: Seems to me like the `BitArray` interface should just have `Serialize` and `Deserialize` methods exposed.

Comment (dustinhiatt-wf) [inline bitarray/sparse_bitarray.go]: Yeah, you could just put (un)marhsal methods on the interface that take the io interfaces, or you could make (un)marshal functions on the package that take the BitArray interface and return byte arrays.  The latter is probably closer to how the standard library marshaling functionality works.

Comment (evanh) [inline bitarray/sparse_bitarray.go]: The problem with the latter choice is that the Unmarshal function would have to return

**Comment by evanh** [comment]
So I changed it up to use a Marshal and Unmarshal functionality instead of the Read/Write functions I had before. Let me know what you think.

**Comment by tylertreat-wf** [comment]
lgtm

**Comment by alexandercampbell-wf** [comment]
+1

**Pull request merged**